test/unittests/pydra/test_engine.py

```diff
@@ -0,0 +1,92 @@
+from distutils import core
```

Comment:
I wonder whether we are testing the right thing here.

From what I read, we are saying the implementation works if nodes named `bids_reader_task` or `bids_writer_task_0` are present anywhere in the workflow. In my opinion, we are testing implementation details rather than actual properties of what the decorator achieves.

If I change the name of the task in the implementation, it does affect that I can still read or write BIDS data in the workflow, yet the test would fail.

An alternative could be to test the reader with a task that simply accumulates paths received sequentially as input from the BIDS reader and check that its results include all paths expected from the BIDS dataset provided as input.

Same for the writer. Making a task that yields a bunch of arbitrary files and checking that the writer wrote all of them to their expected location.

What do you think?

Reply:
Yes I agree and I actually hesitated to write out tests for these functions. In reality, since the interfaces are bound to change (not within the scope of this PR). the only reason these tests exists is to preserve the good practice of having unit tests.
Once the `bids_reader_task` and the `bids_writer_task` are actually written to what we want them to achieve, what you are suggesting will make sense.

Reply:
I updated the description of the PR to reflect that point.